code-dot-org/code-dot-org
# Add Make a Copy feature for Resources

Add a dropdown button to resources that have a google docs link, including options to download the resource as a PDF, as a MSO document, or to make a copy in Google Docs. 

![image](https://user-images.githubusercontent.com/1382374/195931814-a9a15016-b3c9-40e3-b7ba-badd0a0f1108.png)

## Links




- jira ticket: [LP-2030](https://codedotorg.atlassian.net/browse/LP-2030)


## Testing story



* Manually test new functionality locally on a lesson overview page, like [Safety in My Online Neighborhood](http://localhost-studio.code.org:3000/s/coursea-2022/lessons/1).
* Compare existing buttons on lesson overview pages between local and prod.
* Compare all the buttons on the [local teacher dashboard](http://localhost-studio.code.org:3000/home) with the [production teacher dashboard](https://studio.code.org/home).
  * Check different button states: `:hover`, `:visited`, etc.
* Check Storybook renders for Button, DropdownButton, and MakerLanding.
* Run storybook tests.
* Manually check the expand/collapse functionality inside the purple dropdown button on http://localhost-studio.code.org:3000/s/allthemigratedthings/lessons/1/student
* Run new/existing JS unit tests: ButtonTest, DropdownButtonTest, and googleDocsUtilsTest.
